src/req.c

```diff
@@ -83,32 +84,23 @@ hsk_dns_req_create(
   // Grab the first question.
   hsk_dns_qs_t *qs = msg->qd.items[0];
 
-#if 0
-  if (qs->class != HSK_DNS_IN)
-    goto fail;
-
-  // Don't allow dirty names.
-  if (hsk_dns_name_dirty(qs->name))
-    goto fail;
-#endif
-
-  // Check for a TLD.
-  hsk_dns_label_get(qs->name, -1, req->tld);
-
-  // Don't allow dirty TLDs.
-  if (hsk_dns_name_dirty(req->tld))
-    goto fail;
+  // Grab the TLD -- still in DNS wire format
+  uint8_t tld[HSK_DNS_MAX_LABEL + 2] = {0};
+  hsk_dns_label_get(qs->name, -1, tld);
 
   // Lowercase.
-  hsk_to_lower(req->tld);
+  hsk_to_lower(tld);
+
+  // Store the TLD inlcuding the leading length byte
+  memcpy(req->tld, &tld, tld[0] + 1);
 
   // Reference.
   req->ns = NULL;
 
   // DNS stuff.
   req->id = msg->id;
   req->labels = hsk_dns_label_count(qs->name);
-  strcpy(req->name, qs->name);
+  memcpy(req->name, qs->name, sizeof(qs->name));
```

Comment:
sizeof will always be HSK_DNS_MAX_NAME.

Reply:
:+1: fixed in f5200ab1251950b897e8ef3f71f5a11737aeaa5a